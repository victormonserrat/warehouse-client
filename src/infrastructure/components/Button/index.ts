import styled from 'styled-components'

const Button = styled.button`
  background-color: ${({ theme }) => theme.colors.blueLight};
  border-radius: 2.5rem;
  color: ${({ theme }) => theme.colors.white};
  font-family: ${({ theme }) => theme.fonts.primary};
  font-size: ${({ theme }) => theme.sizes.small};
  font-weight: bold;
  padding: 1.25rem 2.5rem;
  width: 100%;
  transition: background-color 0.3s ease;

  :hover,
  :focus {
    background-color: ${({ theme }) => theme.colors.blue};
  }

  :active {
    background-color: ${({ theme }) => theme.colors.blueDark};
  }

  :disabled {
    background-color: ${({ theme }) => theme.colors.white};
    box-shadow: inset 0 0 0 0.0625rem ${({ theme }) => theme.colors.greyLight};
    color: ${({ theme }) => theme.colors.greyLight};
  }
`
export default Button
