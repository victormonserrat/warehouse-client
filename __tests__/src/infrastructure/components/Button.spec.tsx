import Button from '~/infrastructure/components/Button'
import { render } from '~/tests/utils'

describe('Button', () => {
  it('renders correctly', () => {
    const { container } = render(<Button>Jest</Button>)

    expect(container).toMatchSnapshot('Regular')
  })

  it('renders disabled correctly', () => {
    const { container } = render(<Button disabled>Jest</Button>)

    expect(container).toMatchSnapshot('Disabled')
  })
})
