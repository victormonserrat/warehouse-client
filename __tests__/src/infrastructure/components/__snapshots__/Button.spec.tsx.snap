// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly: Regular 1`] = `
<div>
  <button
    class="sc-bczRLJ cpcsKP"
  >
    Jest
  </button>
</div>
`;

exports[`Button renders disabled correctly: Disabled 1`] = `
<div>
  <button
    class="sc-bczRLJ cpcsKP"
    disabled=""
  >
    Jest
  </button>
</div>
`;
